core/src/main/java/org/opentosca/toscana/core/plugin/AbstractPlugin.java

```diff
@@ -1,56 +1,48 @@
 package org.opentosca.toscana.core.plugin;
 
 import org.opentosca.toscana.core.transformation.platform.Platform;
-import org.opentosca.toscana.core.transformation.properties.Property;
+
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.util.HashSet;
-
 /**
- * This class describes a Abstract plugin and implements some things that you would have to implement
- * manually when using the plugin interface
+ * This class describes a Abstract plugin and implements some things that you would have to implement manually when
+ * using the plugin interface
  */
 public abstract class AbstractPlugin implements TransformationPlugin {
 
     /**
      * This logging object can be used for non transformation specific logging within the subclasses
      */
     protected Logger logger = LoggerFactory.getLogger(getClass());
+    private final Platform platform;
 
-    public AbstractPlugin() {
+    public AbstractPlugin(Platform platform) {
+        this.platform = platform;
+        if (platform == null) {
+            throw new IllegalStateException("initPlatform() must return the supported Platform of the plugin.");
+        }
         this.init();
-        logger.info("Initialized '{}' Plugin.", getName());
+        logger.info("Initialized '{}' Plugin.", platform.name);
```

Comment:
Maybe we should hide this information. Would probably look better. if thats not done already. this attribute should at least be final to prevent modification